frontend/public/components/utils/truncate-middle.ts

```diff
@@ -0,0 +1,13 @@
+const ELLIPSIS: string = '\u2026';
+
+// Truncates a string down to `maxLength` characters by replacing the middle portion of the string with an ellipsis
+export const truncateMiddle = (text: string, maxLength: number = 20): string => {
```

Comment:
`ELLIPSIS` is a single character, but the code is assuming 3 characters (for `...`), so the returned string ends up being 2 characters shorter than `maxLength`.

Because of the `Math.floor`, when `maxLength` is odd, the truncated string ends up 1 character shorter than it should be. e.g. `truncateMiddle('test string', 4)` and `truncateMiddle('test string', 5)` return the same string.

I think we want:
```
if (text.length <= maxLength || maxLength < ELLIPSIS.length) {
  return text;
}
const startLength = Math.ceil((maxLength - ELLIPSIS.length) / 2);
const endLength = maxLength - startLength - ELLIPSIS.length;
const startFragment = text.substr(0, startLength);
const endFragment = text.substr(text.length - endLength);
return `${startFragment}${ELLIPSIS}${endFragment}`;
```

Reply:
@kyoto I took it a step further and made this function look and behave a bit more like the lodash `truncate` function.

Reply:
👍